infrastructure/src/main/java/org/corfudb/infrastructure/CorfuServer.java

```diff
@@ -198,38 +57,54 @@ public static void main(String[] args) {
         }
     }
 
-    private static void startServer(String[] args) {
-        // Parse the options given, using docopt.
-        Map<String, Object> opts = new Docopt(USAGE)
-                .withVersion(GitRepositoryState.getRepositoryState().describe)
-                .parse(args);
+    private static void startServer(String[] args) throws ParseException {
```

Comment:
can we have a separate PR for cmd line refactoring?

Reply:
The point of this PR is to load command lines into a Configuration type, what's the point of doing them in two steps ?

If we split this over two PRs, then we need to have a converter from Map<String, Obect> to ServerConfigurations and then another patch to remove that an remplace it with Apache CLI options?